### ISSUE TITLE
Many bugs, handle it!

### ISSUE THREAD

**Issue by Bahzur** [Epix37/Hearthstone-Deck-Tracker#520]
Dont take the header to seriously, its just a reference to this: https://www.youtube.com/watch?v=0yKqKg6DfTY

So, after the newest release the deck tracker got really unstable for me. Things that occured:
-The Tracker does not recognize the match is finished, even when i arrive in the menu
-The cards my opponent used also were not reset that way, they stayed in cardhistory in the next match
-2 matches were not recorded even though it seemed the tracker noticed the match ended
-The note popup after the match minimized Hearthstone, and i wasnt able to open Hearthstone after that (Sound was still there, but even the windowpreview from Win7 didnt show anything and no clicking helped)
![bugseverywhere](https://cloud.githubusercontent.com/assets/4956200/5607924/b64c196e-946e-11e4-9477-40f8673fdda1.png)
-I then turned the "wait until in menu" option on for the notedialog, but when i wrote my note into it, everytime i pressed enter or "set", the dialog just popped up again
-This gets spammed in the log: CURRENTLY NO CONTROLLER SET FOR CARD, WAITING...
-I just noticed, while i have that minimizing bug, everytime i try to open Hearthstone again this gets spammed: Refreshed overlay topmost status
Heres the log:http://pastebin.com/75yKzsRK

**Comment by Bahzur** [comment]
Basically, its so broken right now that i cant play Hearthstone with the tracker at the moment. :(

**Comment by Epix37** [comment]
1-3 and 7 are effects of the same issue (see #511), fixing 4 right now, 5 is not a bug, that was just a debugging message for me, and 6: is that in any way (outside the log) noticable

**Comment by Bahzur** [comment]
No, it just prevented me from maximising Hearthstone again. The line "Refreshed overlay topmost status" was the only thing i noticed when i looked at the in-program-log which appeared everytime when i clicked on the Hearthstone task in the taskbar to maximize it again.

**Comment by Epix37** [comment]
So do you have to close the tracker / restart hs to maximize is again?

**Comment by Bahzur** [comment]
Yup, complete HS restart.

**Comment by Epix37** [comment]
Does it still happen if you enable "hide in menu" in options > general > overlay?

**Comment by Bahzur** [comment]
Yes, i always enable that option. Also, that minimize bug only happened when i didnt choose the option for the notedialog "Wait until back in menu" (but then the notedialog didnt disappear after setting the note which i assume was because of the not noticing end-of-match bug. The overlay wasnt hidden too when i choose that option).

**Comment by Epix37** [comment]
v0.7.2 should fix numbers 1-4 and 7

**Comment by dekow** [comment]
My deck didn't reset after a game. Had to restart the decktracker mid game. (v0.7.2)

**Comment by Bahzur** [comment]
Thx! Will test that version later.

**Comment by Bahzur** [comment]
@dekow As Epix37 mentioned, that should be fixed with that new version, test that.

**Comment by dekow** [comment]
I have the latest version. It still happens!

**Comment by Bahzur** [comment]
Version 0.7.2?

**Comment by dekow** [comment]
Yes.

**Comment by Bahzur** [comment]
Strange. :(
But youll need to wait for an answer from Epix37, I dont have any clue what exactly is going wrong.

**Comment by Epix37** [comment]
v0.7.3 should *actually* fix this now.
